llvm/include/llvm/CodeGen/RegAllocEvictionAdvisor.h

```diff
@@ -191,13 +193,66 @@ class RegAllocEvictionAdvisorAnalysis : public ImmutablePass {
   const AdvisorMode Mode;
 };
 
+/// Common provider for legacy and new pass managers.
+/// This keeps the state for logging, and sets up and holds the provider.
+/// The legacy pass itself used to keep the logging state and provider,
+/// so this extraction helps the NPM analysis to reuse the logic.
+class RegAllocEvictionAdvisorProvider {
```

Comment:
could you add a TODO here saying this is a point in time artifact for the period LPM and NPN coexist, after which this should be collapsed into the NPM analysis?